jupyter_publishing_service/authenticator/hub.py

```diff
@@ -0,0 +1,25 @@
+from httpx import AsyncClient
+from traitlets import Unicode
+from traitlets.config import LoggingConfigurable
+
+from jupyter_publishing_service.constants import JUPYTERHUB_SCOPE
+
+from .abc import AuthenticatorABC
+
+
+class HubAuthenticator(LoggingConfigurable):
+
+    hub_url = Unicode(default_value="http://localhost:8000").tag(config=True)
+
+    async def authenticate(self, user_token):
```

Comment:
For sharing across hubs, the receiving user may not be a Hub user or may not belong to the same hub. I think it is best to use the USER table that comes with publishing service for this. 

I think it's best to consider an anonymous user for Hub login, and then later check with publishing database.